File: src/LightNap.Core/StaticContents/Models/StaticContentUserVisibility.cs
namespace LightNap.Core.StaticContents.Models
{
    /// <summary>
    /// Describes the visibility level of static content to the current user.
    /// </summary>
    public enum StaticContentUserVisibility
    {
        /// <summary>
        /// The user must authenticate before accessing the content.
        /// </summary>
        RequiresAuthentication = 0,

        /// <summary>
        /// The user is not authorized to access the content.
        /// </summary>
        Restricted = 1,

        /// <summary>
        /// User has read-only access to the content.
        /// </summary>
        Reader = 2,

        /// <summary>
        /// User has full edit access to the content.
        /// </summary>
        Editor = 3,
    }
}

File: src/lightnap-ng/src/app/core/backend-api/static-content-user-visibility.ts
/*
 * Static content visibilities for the current user's status.
 */
export type StaticContentUserVisibility = "RequiresAuthentication" | "Restricted" | "Reader" | "Editor";

export const StaticContentUserVisibilities = {
  RequiresAuthentication: "RequiresAuthentication",
  Restricted: "Restricted",
  Reader: "Reader",
  Editor: "Editor",
} as const;
